### PR TITLE
Cleans up some cult code

### PR DESCRIPTION
**Pull request by ChangelingRain**
Notable changes:
Cult space suit is actually a hardsuit now, no inbuilt light, though.
The shielded cult suit has slightly more appropriate visuals.
Constructs that aren't cultists can open cult doors.(this is basically a harvester fix)
Touching a cult door as a noncultist is like 50% spookier.
Veil Shifter will re-pull whatever you were pulling when you used it.(and it'll teleport non-mob objects you were pulling, too)
Veil Shifter won't use a charge if it fails to find a destination.
Fixes a few abuse cases, including multiple items from one item creator at once, hulks being able to pick up cult shit, and farming infinite runed metal via girders.
Artificer walls don't give runed metal or produce a girder.

Less-notable changes:
Fixed hooded cult shit disappearing with the hood up.
Anybody can CLOSE a cult door, but if you're not a cultist this is less than useful.
Veil Shifters leave a useless depowered shifter instead of being deleted.

shield

### CODE REVIEW

**Comment by coiax** [comment]
50% spookier is too op, change to 49% spookier plox

**Comment by KazeEspada** [comment]
Better question, Why aren't harvestors cultists?

**Comment by ChangelingRain** [comment]
I believe it was so that the round end report wasn't five thousand harvesters.

**Comment by KorPhaeron** [comment]
It was that yeah.

Also so harvesters are not confused about being on a team with the human cultists (they're not)

**Comment by KazeEspada** [comment]
It always bugged me that I couldn't identify cultists or commune with fellow harvestors. "HoS, walled up on shuttle. Got gas masks and armory. Need backup."

File: code/game/gamemodes/cult/cult_items.dm [hunk 1]
```diff
@@ -284,48 +282,58 @@
...
+		user << "<span class='cult'>It seems drained.</span>"
+
 /obj/item/device/cult_shift/proc/handle_teleport_grab(turf/T, mob/user)
-	var/mob/living/carbon/human/H = user
-	if(H.pulling && (istype(H.pulling, /mob/living)))
-		var/mob/living/victim =	H.pulling
-		if(!victim.anchored)
-			victim.forceMove(locate(T.x+rand(-1,1),T.y+rand(-1,1),T.z))
-	return
+	. = null
```
Comment (phil235) [inline code/game/gamemodes/cult/cult_items.dm]: I don't think it's needed, procs return null by default I think.

**Comment by ChangelingRain** [comment]
With just how *many* harvesters there are a comms channel would be less than useful.
Besides, it's an endgame thing, it doesn't matter so much if it's clunky to play because it's not part of normal rounds.

**Comment by KorPhaeron** [comment]
>It always bugged me that I couldn't identify cultists

That's the point, you're supposed to just eat everyone

**Comment by KorPhaeron** [comment]
I am not sure sparkles is the best idea for shield. It's good to have some visual consistency so players will know at a glance that it will block something.

**Comment by Robustin** [comment]
Why leave a depowered veil shifter behind? I feel like that would just confuse cultists who pick it up and expect it to do something, at best cluttering a cult base.

**Comment by ChangelingRain** [comment]
It's very visually different and yeah, it exists for clutter and fingerprints if the detective can grab one(not like that's likely to matter but hey)

**Comment by Robustin** [comment]
Joan take out all runed metal from wall/girder deconstruction. I tried to get it all in my PR but I missed Tome-Deconstruction (didnt even know that was a feature). Since artificers can spam walls, ANY runed metal from wall-decon is a bad thing.

Also I'm adding an icon fix to your PR, I uploaded the wrong file when trying to fix merge conflicts.

**Comment by ChangelingRain** [comment]
I'll do a slightly different fix; artificer walls won't drop runed metal instead.
Which file did you upload wrong, I can fix it pretty easy(if I haven't already, I fixed the missing hooded suit sprites)

**Comment by Robustin** [comment]
icons/mob/suits.dmi

I had it all working too but the _t stuff was lost in the final merge conflict.

**Pull request merged**